### PR TITLE
Add a fetch by id api

### PR DESCRIPTION
**Pull request by BenHolden010**
## What I did:
made a new api for individual poems
### Did this break anything?

- [x] No
- [ ]  Yes

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -25,6 +25,15 @@ app.get('/api/v1/poems', (req, res) => {
     res.json({ poems })
 })
 
+app.get('/api/v1/poems/:id', (req,res)=> {
+    const poem = app.locals.poems.find(poem=>poem.id===req.params.id)
+    if(!poem){
+     return res.status(404).json({error:'You yearn for a rhyme but your request ran out of time. please return home.'})
```
Comment (Scotty-Brown) [inline server.js:31]: great job with 404 status code

File: server.js [hunk 2]
```diff
@@ -25,6 +25,15 @@ app.get('/api/v1/poems', (req, res) => {
...
+    } else {
+        res.status(200).json({poem})
+    }
+})
+
```
Comment (Denzel-Turambi) [inline server.js:36]: Nice job implementing 404 error message

**Review by Denzel-Turambi — APPROVED** [review]

File: server.js [hunk 3]
```diff
@@ -25,6 +25,15 @@ app.get('/api/v1/poems', (req, res) => {
     res.json({ poems })
 })
 
+app.get('/api/v1/poems/:id', (req,res)=> {
+    const poem = app.locals.poems.find(poem=>poem.id===req.params.id)
+    if(!poem){
+     return res.status(404).json({error:'You yearn for a rhyme but your request ran out of time. please return home.'})
+    } else {
+        res.status(200).json({poem})
```
Comment (Scotty-Brown) [inline server.js:33]: great job with the 200 status code

**Review by Scotty-Brown — APPROVED** [review]
@Denzel-Turambi @LDeLaRosa13 @BenHolden010 
With this branch that Ben did, he added status codes for the fetch get request.
I think we will need to go back and add status codes for our other get and post request.

**Comment by Denzel-Turambi** [comment]
> @Denzel-Turambi @LDeLaRosa13 @BenHolden010 With this branch that Ben did, he added status codes for the fetch get request. I think we will need to go back and add status codes for our other get and post request.

Yea, I was thinking the same thing!

**Review by LDeLaRosa13 — APPROVED** [review]
Looks great!

**Pull request merged**